cassandra/cluster.py

```diff
@@ -2143,9 +2160,10 @@ def _create_response_future(self, query, parameters, trace, custom_payload, time
                     "BatchStatement execution is only supported with protocol version "
                     "2 or higher (supported in Cassandra 2.0 and higher).  Consider "
                     "setting Cluster.protocol_version to 2 to support this operation.")
+            statement_keyspace = query.keyspace if ProtocolVersion.uses_keyspace_flag(self._protocol_version) else None
```

Comment:
We need to compare here because a BoundStatment always has a keyspace set and this would raise an Exception under V5 protocol.